api/pkg/cmd/stack.go

```diff
@@ -44,18 +49,43 @@ func (s Stack) GetAppStacks(ctx context.Context, payload model.AppStackPayload)
 	return nil, nil
 }
 
-func convertParamToStacklist(paramOutput string, payload model.AppStackPayload) ([]*model.AppStack, error) {
+func parseParamToStacklist(paramOutput string) ([]string, error) {
 	var stacklist []string
 	err := json.Unmarshal([]byte(paramOutput), &stacklist)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not parse json")
-	}
+	return stacklist, errors.Wrap(err, "could not parse json")
+}
+
+func enrichStacklistMetadata(ctx context.Context, stacklist []string, payload model.AppStackPayload, integrationSecret *config.IntegrationSecret) ([]*model.AppStackResponse, error) {
+	workspaceRepo := workspace_repo.NewWorkspaceRepo(
+		integrationSecret.Tfe.Url,
+		integrationSecret.Tfe.Org,
+	).WithTFEToken(setup.GetConfiguration().TFE.Token)
+	wg := sync.WaitGroup{}
+
+	stacks := []*model.AppStackResponse{}
 
-	stacks := []*model.AppStack{}
 	for _, stackName := range stacklist {
-		appStack := model.MakeAppStack(payload.AppName, payload.Environment, stackName)
-		stacks = append(stacks, &appStack)
+		wg.Add(1)
+		go func(stackName string) {
+			stack := &model.AppStackResponse{
+				AppMetadata: *model.NewAppMetadata(payload.AppName, payload.Environment, stackName),
+			}
+			workspace, err := workspaceRepo.GetWorkspace(ctx, fmt.Sprintf("%s-%s", payload.AppMetadata.Environment, stackName))
+			if err == nil {
+				stack.WorkspaceUrl = workspace.GetWorkspaceUrl()
+				stack.Endpoints = map[string]string{}
+				endpoints, err := workspace.GetEndpoints(ctx)
+				if err == nil {
+					stack.Endpoints = endpoints
+				}
+				stack.WorkspaceStatus = workspace.GetCurrentRunStatus(ctx)
+			}
+			stacks = append(stacks, stack)
+
+			wg.Done()
```

Comment:
make this a defer at the top of the function so we don't have to worry about changes to this code exiting early and not signaling the waiter should stop waiting

Reply:
good call, updated